File: spring-microservices/course-service/src/main/java/com/hoangphihiep/dto/response/CourseProgressDetailResponse.java
package com.hoangphihiep.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CourseProgressDetailResponse {
    private CourseProgressResponse courseProgress;
    private int totalItems;
    private int completedItems;
    private int completedLessons;
    private int completedQuizzes;
    private int completedAssignments;
}

File: react-type-vite/src/services/api/response/courseProgressDetailResponse.ts
import type { CourseProgressResponse } from "./courseProgressResponse";

export interface CourseProgressDetailResponse {
  courseProgress: CourseProgressResponse;
  totalItems: number;
  completedItems: number;
  completedLessons: number;
  completedQuizzes: number;
  completedAssignments: number;
}